### ISSUE TITLE
Unhandled error for request PUT /Operadores/141: Error: id property (id) cannot be updated from 141 to 141

### ISSUE THREAD

**Issue by lainz** [strongloop/loopback#3872]
Hi, when migrating to loopback 3 I now get this error message, on a PUT that worked before:

```
Unhandled error for request PUT /Operadores/141: Error: id property (id) cannot be updated from 141 to 141
    at Function.DataAccessObject.replaceById (.\node_modules\loopback-datasource-juggler\lib\dao.js:3023:11)
    at SharedMethod.invoke (.\node_modules\strong-remoting\lib\shared-method.js:270:25)
    at HttpContext.invoke (.\node_modules\strong-remoting\lib\http-context.js:297:12)
    at phaseInvoke (.\node_modules\strong-remoting\lib\remote-objects.js:685:9)
    at runHandler (.\node_modules\loopback-phase\lib\phase.js:135:5)
    at iterate (.\node_modules\loopback-phase\node_modules\async\lib\async.js:146:13)
    at Object.async.eachSeries (.\node_modules\loopback-phase\node_modules\async\lib\async.js:162:9)
    at runHandlers (.\node_modules\loopback-phase\lib\phase.js:144:13)
    at iterate (.\node_modules\loopback-phase\node_modules\async\lib\async.js:146:13)
    at .\node_modules\loopback-phase\node_modules\async\lib\async.js:157:25
    at .\node_modules\loopback-phase\node_modules\async\lib\async.js:154:25
    at execStack (.\node_modules\strong-remoting\lib\remote-objects.js:530:7)
    at RemoteObjects.execHooks (.\node_modules\strong-remoting\lib\remote-objects.js:534:10)
    at phaseBeforeInvoke (.\node_modules\strong-remoting\lib\remote-objects.js:681:10)
    at runHandler (.\node_modules\loopback-phase\lib\phase.js:135:5)
    at iterate (.\node_modules\loopback-phase\node_modules\async\lib\async.js:146:13)
```

JSON:

```
  "id": {
    "type": "String",
    "id": 1,
    "required": true,
    "length": null,
    "precision": null,
    "scale": null,
    "postgresql": {
      "columnName": "id",
      "dataType": "uuid",
      "dataLength": null,
      "dataPrecision": null,
      "dataScale": null,
      "nullable": "NO"
    }
  }
```

**Comment by jannyHou** [comment]
hi @lainz there is a migration guide from loopback 2 to loopback 3, see https://loopback.io/doc/en/lb3/Migrating-to-3.0.html

In your case, the error is probably related to the breaking change of verb `PUT`.
TL;DR, in loopback3, the verb `PUT` means a completed replace not a partial update anymore:

- loopback2
  - PUT --> partial update

- loopback3
  - PUT --> replace
  - PATCH --> partial update

For details, please see https://loopback.io/doc/en/lb3/Migrating-to-3.0.html.

**Comment by lainz** [comment]
Thanks, it solves one of the several problems we had when migrating.
